experiment/secret-sync-controller/secret_sync_spec.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"context"
+	// b64 "encoding/base64"
+	// "bytes"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"strconv"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretSyncSpec struct {
+	// the target can be either a K8s secret or a SecretManager secret
+	Source      Target `yaml:"source"`
+	Destination Target `yaml:"destination"`
+}
+
+type Target struct {
+	// assert that one of the two should be `nil`
+	Kubernetes    *KubernetesSpec    `yaml:"kubernetes"`
+	SecretManager *SecretManagerSpec `yaml:"secretManager"`
+}
+
+type KubernetesSpec struct {
+	Namespace string `yaml:"namespace"`
+	Secret    string `yaml:"secret"`
+}
+
+type SecretManagerSpec struct {
+	Project string `yaml:"project"`
+	Secret  string `yaml:"secret"`
+}
+
+func (target Target) GetLatestSecretVersion(k8s_clientset *kubernetes.Clientset, secretManager_ctx context.Context, secretManager_client *secretmanager.Client) (int, map[string][]byte) {
```

Comment:
snake_case isn't idiomatic go, prefer camelCase
```suggestion
func (target Target) GetLatestSecretVersion(k8sClientSet *kubernetes.Clientset, secretManagerCtx context.Context, secretManagerClient *secretmanager.Client) (int, map[string][]byte) {
```

Reply:
I would instead make `ctx context.Context` your first parameter, and plumb that through (instead of making a `context.TODO()` in the function below)

Reply:
- Obtained BUILD.bazel ("experiment/BUILD.bazel", "experiment/secret-sync-controller/BUILD.bazel", "experiment/secret-sync-controller/client/BUILD.bazel")
- Renamed all snake case variables to camel case.
- The context argument is removed after switching to k8s.io/client-go v0.17.3